выполнение проекта 5го спринта

Aleksandra1505 opened the pull request:
выполнение финального задания 5го спринта

Aleksandra1505 committed:
выполнение проекта 5го спринта

AlexKitIt commented on test/locators.py:
```diff
@@ -0,0 +1,26 @@
+from selenium.webdriver.common.by import By
+
+# Локаторы для прохождения регистрации
+NAME_INPUT = (By.XPATH, "//form[@class = 'Auth_form__3qKeq mb-20' ]//fieldset[1]//input")  # Поле для ввода имени
```
Нужно исправить: здесь и далее: использование порядкового номера элемента в dom является весьма нестабильным способом поиска, необходимо привязываться к атрибутам конкретного элемента
Кроме того ходить от form и root и использовать абсолютные пути нельзя - очень ненадежно, добавится на пути новый элемент - тест ляжет. Все подобные локаторы необходимо переписать

AlexKitIt commented on test/test_account_logout.py:
```diff
@@ -0,0 +1,31 @@
+from locators import LOGIN_EMAIL_INPUT, LOGIN_PASSWORD_INPUT, LOGIN_BUTTON_MAIN_PAGE, LOGIN_BUTTON_AUTH_PAGE, PROFILE_LINK, LOGOUT_BUTTON
+import time
+
+driver = webdriver.Chrome()
```
Нужно исправить: здесь и далее: инстанс драйвера должен подниматься и закрываться в фикстуре

AlexKitIt commented on test/test_account_logout.py:
```diff
@@ -0,0 +1,31 @@
+import time
+
+driver = webdriver.Chrome()
+driver.get("https://stellarburgers.nomoreparties.site/")
```
Можно лучше: урлы стоит хранить отдельно, они постоянно переиспользуются, для них стоит создать модуль urls или отдельный класс в data

AlexKitIt commented on test/test_account_logout.py:
```diff
@@ -0,0 +1,31 @@
+
+driver = webdriver.Chrome()
+driver.get("https://stellarburgers.nomoreparties.site/")
+
```
Нужно исправить: здесь и далее:  это должны быть не исполняемые модули а тестовые методы, размещенные в тестовые классы, объединенные в них по общей тестируемой функциональности

AlexKitIt commented on test/test_account_logout.py:
```diff
@@ -0,0 +1,31 @@
+
+# Авторизация
+driver.find_element(*LOGIN_BUTTON_MAIN_PAGE).click()
+driver.find_element(*LOGIN_EMAIL_INPUT).send_keys("antropova15@yande.ru")
+driver.find_element(*LOGIN_PASSWORD_INPUT).send_keys("111111")
```
Можно лучше: тестовые данные стоит хранить отдельно в data модуле, так поддерживать и переиспользовать будет проще, в тестах и тестовых классах их держать не стоит

AlexKitIt commented on test/test_account_logout.py:
```diff
@@ -0,0 +1,31 @@
+driver.find_element(*PROFILE_LINK).click()
+WebDriverWait(driver, 10).until(EC.url_to_be("https://stellarburgers.nomoreparties.site/account/profile"))
+assert "https://stellarburgers.nomoreparties.site/account/profile" in driver.current_url, "Не удалось перейти в личный кабинет."
+time.sleep(2)
```
Нужно исправить: здесь и далее: слипов и неявных ожиданий быть не должно. Необходимо использовать expected_conditions и встраивать ожидания на нем.

AlexKitIt commented on test/test_ingredients_navigation.py:
```diff
@@ -0,0 +1,40 @@
+import time
+
+#Проверка перехода к заданному разделу
+def check_section_navigation(driver, section_locator):
+    driver.find_element(*section_locator).click()
```
Нужно исправить: здесь и далее: в тестовых модулях хранятся только тестовые методы, в рамках данного проекта этот шаг стоит хранить прямо в тесте, потом расскажут куда его лучше положить

AlexKitIt commented on test/test_ingredients_navigation.py:
```diff
@@ -0,0 +1,40 @@
+def test_navigation_to_fillings():
+    driver = webdriver.Chrome()
+    driver.get("https://stellarburgers.nomoreparties.site/")
+    check_section_navigation(driver, CONSTRUCTOR_FILLINGS_SECTION)
+    time.sleep(2)
```
Нужно исправить: здесь и далее: тест всегда заканчивается assert где происходит сравнение ожидаемого результата с фактическим. Тут его нет, проверку необходимо дописать

AlexKitIt commented on test/test_registration.py:
```diff
@@ -0,0 +1,49 @@
+
+    driver.find_element(*NAME_INPUT).send_keys("Сашаа")
+    time.sleep(2)
+    driver.find_element(*EMAIL_INPUT).send_keys("antropova4155645@yandex.ru")
```
Нужно исправить: для теста регистрации email должен генерироваться каждый раз новый, в противном случае мы получим ошибку уже созданного пользователя. Необходимо написать метод, генерирующий email и положить его в helpers

AlexKitIt commented on test/test_registration.py:
```diff
@@ -0,0 +1,49 @@
+    error_message = driver.find_element(*ERROR_MESSAGE)
+    assert error_message.is_displayed(), "Ошибка: сообщение об ошибке не отображается."
+    assert error_message.text == "Некорректный пароль", "Ошибка: текст сообщения не совпадает"
+    print("Тест на некорректный пароль пройден.")
```
Нужно исправить: принтов быть не должно

AlexKitIt commented on test/tests_authentication.py:
```diff
@@ -0,0 +1,48 @@
+driver = webdriver.Chrome()
+
+# 1. Вход по кнопке "Войти в аккаунт" на главной
+driver.get("https://stellarburgers.nomoreparties.site/")
+driver.find_element(*LOGIN_BUTTON_MAIN_PAGE).click()
+login(driver, "antropova15@yande.ru", "111111")
+check_order_button_displayed(driver)
+driver.quit()
+
+# 2. Вход через кнопку "Личный кабинет"
+driver = webdriver.Chrome()
+driver.get("https://stellarburgers.nomoreparties.site/")
+driver.find_element(*PROFILE_LINK).click()
+login(driver, "antropova15@yande.ru", "111111")
+check_order_button_displayed(driver)
+driver.quit()
+
+# 3. Вход через кнопку в форме регистрации
+driver = webdriver.Chrome()
+driver.get("https://stellarburgers.nomoreparties.site/register")
+driver.find_element(By.XPATH, '//*[@id="root"]/div/main/div/div/p/a').click()
+login(driver, "antropova15@yande.ru", "111111")
+check_order_button_displayed(driver)
+driver.quit()
+
+# 4. Вход через форму восстановления пароля
+driver = webdriver.Chrome()
+driver.get("https://stellarburgers.nomoreparties.site/forgot-password")
+driver.find_element(By.XPATH, '//*[@id="root"]/div/main/div/div/p/a').click()
+login(driver, "antropova15@yande.ru", "111111")
+check_order_button_displayed(driver)
+driver.quit()
```
Что это за код и какое отношение он имеет к проекту?

Aleksandra1505 committed:
Обновила локаторы для регистрации, авторизации, перехода в конструктор и для переходов по разделам

Aleksandra1505 committed:
в рамках проверки test_registration были внесены следующие правки:
- создан helpers, в него добавлен метод, генерирующий эмейлы
- метод генерирующий эмейлы был добавлен в обе проверки регистрации
- убран принт в тесте

тесты запустила - все работает

Aleksandra1505 committed:
в рамках проверки test/test_ingredients_navigation.py были внесены следующие правки:
- добавлены новые локаторы
- методы перенесены в каждый тест отдельно
- добавлены ассерты

тесты запустила - все работает

Aleksandra1505 committed:
в рамках теста test/test_account_logout.py было исправлено следующее:
- создан тестовый класс TestLogout
- тест test_logout в целом был полностью переписан для соответствия рекомендациям
- убрала time.sleep и взамен добавила WebDriverWait
- создала отдельные файлы data и urls, куда вынесла тестовые файлы
- был создан файл conftest для фикстур, сама фикстура используется в тестах

Aleksandra1505 committed:
финальные правки следующие:
- добавлены новые локаторы и переписаны некоторые старые локаторы
- добавлены новые ссылки в файл urls
- все тесты были переписаны в соответствии с требованиями:
1) инстанс драйвера поднимается и закрыватся в фикстуре
2) урлы хранятся отдельно
3) все тесто оформлены в виде методов внутри класса
4) тестовые данные теперь хранятся в data модуле
5) все слипы были удалены

Aleksandra1505 committed:
Добавлены все файлы в репозиторий

Aleksandra1505 committed:
Удалены файлы конфигурации IDE из репозитория

AlexKitIt commented on test/test_account_logout.py:
```diff
@@ -0,0 +1,30 @@
+from data import LOGIN_EMAIL, LOGIN_PASSWORD
+from locators import LOGIN_EMAIL_INPUT, LOGIN_PASSWORD_INPUT, LOGIN_BUTTON_AUTH_PAGE, PROFILE_LINK, LOGOUT_BUTTON
+
+@pytest.mark.usefixtures("driver")
```
Это можно не использовать, driver и так передается в каждый тест
